payjoin-cli/src/app/v2.rs

```diff
@@ -123,60 +135,79 @@ impl App {
-            .process_v2_proposal(res)
-            .map_err(|e| anyhow!("Failed to process proposal {}", e))?;
-        let (req, ohttp_ctx) = payjoin_proposal
-            .extract_v2_req()
-            .map_err(|e| anyhow!("v2 req extraction failed {}", e))?;
-        println!("Got a request from the sender. Responding with a Payjoin proposal.");
-        let http = http_agent()?;
-        let res = http
-            .post(req.url)
-            .header("Content-Type", payjoin::V2_REQ_CONTENT_TYPE)
-            .body(req.body)
-            .send()
-            .await
-            .map_err(map_reqwest_err)?;
-        payjoin_proposal
-            .process_res(res.bytes().await?.to_vec(), ohttp_ctx)
-            .map_err(|e| anyhow!("Failed to deserialize response {}", e))?;
-        let payjoin_psbt = payjoin_proposal.psbt().clone();
-        println!(
-            "Response successful. Watch mempool for successful Payjoin. TXID: {}",
-            payjoin_psbt.extract_tx().clone().txid()
-        );
-        self.db.clear_recv_session()?;
-        Ok(())
+        let mut interrupt = self.interrupt.clone();
+        tokio::select! {
+            res = self.long_poll_fallback(&mut session) => {
+                let res = res?;
+                println!("Fallback transaction received. Consider broadcasting this to get paid if the Payjoin fails:");
+                println!("{}", serialize_hex(&res.extract_tx_to_schedule_broadcast()));
+                let mut payjoin_proposal = self
+                    .process_v2_proposal(res)
+                    .map_err(|e| anyhow!("Failed to process proposal {}", e))?;
+                let (req, ohttp_ctx) = payjoin_proposal
+                    .extract_v2_req()
+                    .map_err(|e| anyhow!("v2 req extraction failed {}", e))?;
+                println!("Got a request from the sender. Responding with a Payjoin proposal.");
+                let http = http_agent()?;
+                let res = http
+                    .post(req.url)
+                    .header("Content-Type", payjoin::V2_REQ_CONTENT_TYPE)
+                    .body(req.body)
+                    .send()
+                    .await
+                    .map_err(map_reqwest_err)?;
+                payjoin_proposal
+                    .process_res(res.bytes().await?.to_vec(), ohttp_ctx)
+                    .map_err(|e| anyhow!("Failed to deserialize response {}", e))?;
+                let payjoin_psbt = payjoin_proposal.psbt().clone();
+                println!(
+                    "Response successful. Watch mempool for successful Payjoin. TXID: {}",
+                    payjoin_psbt.extract_tx().clone().txid()
+                );
+                self.db.clear_recv_session()?;
+                Ok(())
+            }
+            _ = interrupt.changed() => {
+                Err(anyhow!("Interrupted. Call `send` with the same arguments to resume this session or `resume` to resume all sessions."))
```

Comment:
This is for the receiver.

```suggestion
                Err(anyhow!("Interrupted. Call the `resume` command to resume all sessions."))
```

Reply:
🙏